fix(docs): improvements to the agents 101 course (#185)
fetchai/docs: pages/guides/agent-courses/introductory-course.mdx

    pyenv versions #Here we verify if it's set up correctly
    ```

### Installing Poetry

As our next step, we will install Poetry which will greatly help us in managing all of the Python specific packages. You can install Poetry run the following in your Zsh terminal:

    ```
    curl -sSL https://install.python-poetry.org | python3 -
    ```

    <Callout type="info" emoji="ℹ️">
      For an extended exploration within poetry please visit their website where you'll be able to quickly resolve any installation related questions. You can find more information [here ↗️](https://python-poetry.org/docs/#installation)
    </Callout>

### Initialising the project

Now that we've got all our necessary tools installed, let's create your working directory and initialize Poetry 🎉

    <Callout type="info" emoji="ℹ️">
      Make sure you're in a path you're comfortable with, check this by typing `pwd` into terminal, ideally you'd see something like `/Users/Jessica/Documents`
    </Callout>

Now, let's create a folder to work in: `mkdir development/agent-demo`

Let's change directory (using `cd` command) into `cd development/agent-demo`. Then, let's check the path with `pwd` command.

> If you're happy with the install location for the project, let's go ahead an initialize Poetry:

Reviewer: ```suggestion
If you're happy with the install location for the project, let's go ahead and initialize Poetry:
```

    Let's update `alice_agent.py` with the above.

>     Decorators are moderately advanced in Python, but in this guide all you need to know is that they're there so that the imported agent python library knows to act in a certain way on the declared functions. To find out more about decorators, take a look as [Primer on Python decorators  ↗️](https://realpython.com/primer-on-python-decorators/).

Reviewer: ```suggestion
    Decorators are moderately advanced in Python, but in this guide all you need to know is that they're there so that the imported agent python library knows to act in a certain way on the declared functions. To find out more about decorators, take a look at [Primer on Python decorators  ↗️](https://realpython.com/primer-on-python-decorators/).
```

#### Printing agent's addresses

> As we have mentioned earlier, every uAgent is identified by two addresses within the uAgents Framework: **uAgent** and **Fetch Network** addresses. You can use the python internal function `print()` to print a string representation of any object:

Reviewer: ```suggestion
Sometimes we just need to see what a value looks like; helpfully python has an inbuilt function `print()`. As we have mentioned earlier, every uAgent is identified by two addresses within the uAgents Framework: **uAgent** and **Fetch Network** addresses, so let's print them to console to see their differences.
```

        message: str
    ```

> We need to define a function for `alice` to send messages to `bob` periodically, to do this we define the `send_message()` function using the `Context` class to make our agent `alice` send a message to `bob` on interval:

Reviewer: have we mentioned context by this point?

Author: We mention it just after the printing an agent's address example. I've added clarity around it in the latest commit.

>     Here, we defined a local http address but you could also define a remote address to allow agent communication over different machines through the internet. Importantly, make sure to add a seed to your agent so you don't have to fund different addresses each time you run your agent.

Reviewer: Have we explained what a seed is yet? (I may have missed it)

Author: We explain what addresses are and link them to our github for a deeper dive. We don't specifically mention what a seed phrase it. I'll add clarity and a redirect for a deeper dive.

        alice.run()
    ```

> Similarly, we also need to define a script for `bob` so to create a remote communication with `alice` agent. Once again let's use the `touch remote_bob.py` command:

Reviewer: You don't tell them to copy the code over 👀

Author: The name, message content, seed phrase, port and decorator are all different. While it's easy to copy and adapt the code it makes more sense for people to write it out. Writing it out also reinforces their knowledge.

Reviewer: Yes, but you still should let them know now is a good time to try. No one is hand writing this stuff out, let them copy!

## Agents and storage 

Agents within the uAgents Framework possess the capability to locally store information in a JSON file, ensuring data retrieval as needed. This storage functionality serves as a fundamental component for agents to maintain a state, recollect prior interactions, and base decisions on historical data. 

The objective behind integrating storage features is to empower agents to preserve and leverage information over time, facilitating the recollection of past interactions and context for more informed decision-making. This capacity to learn from past experiences enables agents to adapt and refine their behavior and decision processes.

>     <Callout type="info" emoji="ℹ️">

Reviewer: We referenced JSON earlier with a link, this seems misplaced.

Author: Removed.
